tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -2930,6 +2930,35 @@
                 g_TestResults[CS_INDEX] = tr;
             }
         }
+]]>
+    </Shader>
+  </ShaderOp>
+  <ShaderOp Name="QuadAnyAll" CS="CS" MS="MS" AS="AS" DispatchX="8" DispatchY="8">
+    <RootSignature>RootFlags(ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT), UAV(u0)</RootSignature>
+    <Resource Name="UAVBuffer0" Dimension="BUFFER" Width="256" InitialResourceState="COPY_DEST" Init="ByName" Flags="ALLOW_UNORDERED_ACCESS" TransitionTo="UNORDERED_ACCESS" ReadBack="true" />
+    <Shader Name="MS66" Target="ms_6_6" EntryPoint="main" Text="@CS"/>
+    <Shader Name="AS66" Target="as_6_6" EntryPoint="main" Text="@CS"/>
+    <Shader Name="CS66" Target="cs_6_6" EntryPoint="main" Text="@CS"/>
+    <Shader Name="MS" Target="ms_6_7" EntryPoint="main" Text="@CS"/>
+    <Shader Name="AS" Target="as_6_7" EntryPoint="main" Text="@CS"/>
+    <Shader Name="CS" Target="cs_6_7" EntryPoint="main">
+    <![CDATA[
+RWStructuredBuffer<int2> Values;
+
+[numthreads(8, 8, 1)]
+void main(uint3 ID: SV_DispatchThreadID) {
```

Comment:
Is this valid as an AS without dispatch or MS without any output (or outputtopology)?